### DISCUSSION TITLE
nameToFavoriteNumber in Lesson 2

### DISCUSSION THREAD

**Discussion by yeahChibyke** [Cyfrin/foundry-full-course-cu#342] [Q&A] [answered]
I am coding along with Patrick, and I am currently in Lesson where he explains mapping using the addPerson function.
So far, with mapping added to that(addPerson) function, it is possible to loop through and find a _favoriteNumber if the _name is provided.
But currently, the _name provided must be case sensitive. How can I provide the _name in a non case sensitive manner and still get the _favoriteNumber?

**Comment by shikhar229169** [comment]
You can follow a particular manner in which you handle the _name.
So, you can say the first letter should be capital, even if the user enters the _name in wrong manner, you will convert it to correct format.
Let's say user entered ```anonymous person```, so you can convert that to ```Anonymous Person```, and then deal with it accordingly.

**Reply by yeahChibyke** [reply]
I will try this out and get back to you on the result.

**Comment by Equious** [comment, marked as answer]
Typically the normalization of user inputs is handled through a front end. There's no `.lower()` or `.upper()` functions in solidity, though there are some fancy repos that just do the hard math for you if you really want!
